src/connector/src/source/reader/reader.rs

```diff
@@ -107,7 +107,7 @@ impl SourceReader {
     /// Refer to `WaitCheckpointWorker` for more details.
     pub async fn create_wait_checkpoint_task(&self) -> ConnectorResult<Option<WaitCheckpointTask>> {
         Ok(match &self.config {
-            ConnectorProperties::PostgresCdc(_prop) => {
+            ConnectorProperties::PostgresCdc(_) | ConnectorProperties::SqlServerCdc(_) => {
```

Comment:
The purpose of wait ckpt task is to make sure the persisted source offset in checkpoint is valid, so we commit the offset to upstream before the ckpt is committed. So that PG won't reclaim the offset before the ckpt is done.
Does sqlserver also needs this guarantee?

Reply:
Thanks for the reminder. Sql Server doesn't need this. Removed.